File: Data-Structure-I/75-MEDIUM-Sort-Colors.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return NULL
     */
    function sortColors(&$nums) {
        $beg = 0; 
        $run = 0;
        $end = count($nums) - 1;
        while($run <= $end){
            if($nums[$run] == 2){
                $tmp = $nums[$run];
                $nums[$run] = $nums[$end];
                $nums[$end] = $tmp;
                $end--;
            }
            else if($nums[$run] == 0){
                $tmp = $nums[$run];
                $nums[$run] = $nums[$beg];
                $nums[$beg] = $tmp;
                $beg++;
                $run++;
            }
            else $run++;
        }
    }
}

File: Data-Structure-I/75-MEDIUM-Sort-Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        beg, run, end = 0, 0, len(nums) - 1;
        while run <= end:
            if nums[run] == 2:
                tmp = nums[end]
                nums[end] = nums[run]
                nums[run] = tmp
                end -= 1
            elif nums[run] == 0:
                tmp = nums[beg]
                nums[beg] = nums[run]
                nums[run] = tmp
                run += 1
                beg += 1
            else:
                run += 1
